core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/arrayfunction/ArrayFunctionExpression.kt

```diff
@@ -0,0 +1,41 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.arrayfunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.ValidType
+
+sealed class ArrayFunctionExpression<T : ValidType>(
+    private val symbol: String,
+    private val array: TypeExpression<ArrayType<T>>,
+    private vararg val args: TypeExpression<out ValidType>,
+    private val extra: TypeExpression<out ValidType>? = null,
+) : TypeExpression<ArrayType<T>>, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
+        val arrayDopeQuery = array.toDopeQuery()
+        val argsDopeQuery = args.map { it.toDopeQuery() }
+        val extraDopeQuery = extra?.toDopeQuery()
+        return DopeQuery(
+            queryString = toFunctionQueryString(symbol, arrayDopeQuery, *argsDopeQuery.toTypedArray(), extra = extraDopeQuery),
+            parameters = arrayDopeQuery.parameters + argsDopeQuery.fold(
+                emptyMap(),
+            ) { argsParameters, field -> argsParameters + field.parameters } + extraDopeQuery?.parameters.orEmpty(),
+        )
+    }
+}
+
+sealed class ArrayFunctionNumberExpression<T : ValidType>(
```

Comment:
Why ArrayFunction**Number**Expression? Shouldn't it then be of type _TypeExpression< NumberType >_?

Reply:
Maybe add some more expresive tests, where you do something like `...WHERE ARRAY_COUNT(...) + 7 < 20` - to check that we really have a number after calling such a function